pandas/core/indexes/period.py

```diff
@@ -839,6 +839,16 @@ def memory_usage(self, deep=False):
             result += self._int64index.memory_usage(deep=deep)
         return result
 
+    def _convert_arr_indexer(self, keyarr):
```

Comment:
why would these not share the impl from the superclass?

Reply:
At first, I Implemented on the `base` But,

> would it be simpler to override DatetimeIndex._convert_arr_indexer and PeriodIndex._convert_arr_indexer?

So I changed. Okay I change again.